doc/api/node_report.md

```diff
@@ -0,0 +1,211 @@
+# node-report
+
+Delivers a JSON-formatted Diagnostic summary, written to a file.
+
+The report is intended for development, test and production use, to capture
+and preserve information for problem determination. It includes JavaScript
+and native stack traces, heap statistics, platform information, resource
+usage etc. With the report option enabled, diagnostic reports can be triggered
+on unhandled exceptions, fatal errors and user signals, in addition to
+triggering programatically through API calls.
+
+The built-in report is supported in Node.js versions from REPLACEME onwards.
+For Node.js versions REPLACEME and below, use its [npm counterpart][] instead.
+
+## Usage
+
+```bash
+node --experimental-report --report-uncaught-exception
+ --report-on-signal --report-on-fatalerror app.js
+```
+
+* `--experimental-report` Enables the diagnostic report feature.
+ In the absence of this flag, use of all other related options will result in 
+ an error.
+ 
+* `--report-uncaught-exception` Enables report to be generated on un-caught
+exceptions. Useful when inspecting JavaScript stack in conjunction with native
+stack and other Runtime environment data.  
+
+* `--report-on-signal` Enables report to be generated upon receiving
+the specified (or predefined) signal to the running Node process. (See below
+on how to modify the signal that triggers the report). Default signal is `SIGUSR2`.
+Useful when report needs to be generated at arbitrary control flow points.
+Application monitors may leverage this feature to collect report at regular
+intervals and plot rich set of internal runtime data to their views.
+
+Signal based report generation is not supported in Windows.
+
+Under normal circumstances, there is no need to modify the report triggering
+signal. However, if Node is embedded in native applications that have
+pre-existing listener to `SIGUSR2` this flag helps to change that without
+affecting the behavior of the embedder.
+
+* `--report-on-fatalerror` Enables the node-report to be triggered on
+fatalerror (internal errors within the Node.js runtime such as out of memory)
+that leads to termination of the application. Useful to inspect various
+diagnostic data elements such as heap, stack, event loop state, resource
+consumption etc. to reason about the fatalerror.
+
+* `--report-directory` Location at which the report will be generated.
+
+* `--report-filename` Name of the file to which the report will be written.
+
+* `--report-signal` Sets or resets the signal for report generation (not
+supported on windows). Default signal is `SIGUSR2`.
+
+* `--report-verbose` Flag that controls additional verbose information on report
+generation.
+
+A report can also be triggered via an API call from a JavaScript application:
+
+```js
+const { report } = require('util');
+report.triggerReport();
+```
+
+The API takes an optional additional argument `filename` which is the name
```

Comment:
```suggestion
This function takes an optional additional argument `filename`, which is the name
```

(Or “method”. I don’t believe “API” is common JS usage for individual functions.)